ledger/eval/eval.go

```diff
@@ -1288,13 +1383,17 @@ func (eval *BlockEvaluator) endOfBlock() error {
 		}
 	}
 
-	err := eval.validateExpiredOnlineAccounts()
-	if err != nil {
+	if err := eval.validateExpiredOnlineAccounts(); err != nil {
+		return err
+	}
+	if err := eval.resetExpiredOnlineAccountsParticipationKeys(); err != nil {
 		return err
 	}
 
-	err = eval.resetExpiredOnlineAccountsParticipationKeys()
-	if err != nil {
+	if err := eval.validateAbsentOnlineAccounts(); err != nil {
```

Comment:
Maybe this is a better area for a consensus param gate, have to think on it more..

Reply:
Ideally once in this function applying to the different absent/suspension calls (including knockofflineAccountslist?)

Reply:
These are gated internally based on the allowed size of the list.  That was how expiration was done and I figured keeping it similar would be clearest.

Reply:
I looked around some, I saw the underlying list getting appended to but not where it's size was limited.

Reply:
https://github.com/jannotti/go-algorand/blob/absenteeism/ledger/eval/eval.go#L1537